packages/hub/node-tests/routes/custodial-wallets-test.ts

```diff
@@ -0,0 +1,179 @@
+import { Server } from 'http';
+import supertest, { Test } from 'supertest';
+import { Client as DBClient } from 'pg';
+import { bootServerForTesting } from '../../main';
+import { Container, Registry } from '../../di/dependency-injection';
+
+const stubNonce = 'abc:123';
+let stubAuthToken = 'def--456';
+let stubTimestamp = process.hrtime.bigint();
+
+class StubAuthenticationUtils {
+  generateNonce() {
+    return stubNonce;
+  }
+  buildAuthToken() {
+    return stubAuthToken;
+  }
+  extractVerifiedTimestamp(_nonce: string) {
+    return stubTimestamp;
+  }
+
+  validateAuthToken(encryptedAuthToken: string) {
+    return handleValidateAuthToken(encryptedAuthToken);
+  }
+}
+
+class StubWyreService {
+  createCustodialWallet(address: string) {
+    return handleCreateWyreWallet(address);
+  }
+}
+
+let stubUserAddress = '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13';
+let stubWyreWalletId = 'WA_L36QRQZUPCR';
+let stubDepositAddress = '0x59faede86fb650d956ca633a5c1a21fa53fe151c'; // wyre always returns lowercase addresses
+let wyreCallCount = 0;
+
+function handleValidateAuthToken(encryptedString: string) {
+  expect(encryptedString).to.equal('abc123--def456--ghi789');
+  return stubUserAddress;
+}
+
+function handleCreateWyreWallet(address: string) {
+  wyreCallCount++;
+  return {
+    status: null,
+    callbackUrl: null,
+    srn: `wallet:${stubWyreWalletId}`,
+    pusherChannel: 'blah',
+    notes: null,
+    balances: {},
+    depositAddresses: {
+      ETH: stubDepositAddress, // eslint-disable-line @typescript-eslint/naming-convention
+    },
+    totalBalances: {},
+    availableBalances: {},
+    savingsReferralSRN: null,
+    name: address,
+    id: stubWyreWalletId,
+    type: 'DEFAULT',
+  };
+}
+
+describe('GET /api/custodial-wallet', function () {
+  let server: Server;
+  let db: DBClient;
+  let request: supertest.SuperTest<Test>;
+
+  this.beforeEach(async function () {
+    let container!: Container;
+    wyreCallCount = 0;
+    server = await bootServerForTesting({
+      port: 3001,
+      registryCallback(registry: Registry) {
+        registry.register('authentication-utils', StubAuthenticationUtils);
+        registry.register('wyre', StubWyreService);
+      },
+      containerCallback(serverContainer: Container) {
+        container = serverContainer;
+      },
+    });
+    let dbManager = await container.lookup('database-manager');
+    db = await dbManager.getClient();
+    await db.query('DELETE FROM custodial_wallets');
+    // Add foil test data to assert we never get this result
+    await db.query(
+      'INSERT INTO custodial_wallets (user_address, wyre_wallet_id, deposit_address) VALUES ($1, $2, $3)',
+      [`foil-${stubUserAddress}`.toLowerCase(), `foil-${stubWyreWalletId}`, `foil-${stubDepositAddress}`]
+    );
```

Comment:
The concept of foil test data is new to me. Is this to prevent the case where `/api/custodial-wallet` is getting a random (the first?) wallet from the db, and not the one we request?

Reply:
yup, that's right.